flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/connector/elasticsearch/table/ElasticsearchDynamicSinkFactoryBaseTest.java

```diff
@@ -227,4 +234,19 @@ public void validateWrongCredential() {
                                                 "")
                                         .build()));
     }
+
+    @Test
+    public void testSinkParallelism() {
+        ElasticsearchDynamicSinkFactoryBase sinkFactory = createSinkFactory();
+        DynamicTableSink sink =
+                sinkFactory.createDynamicTableSink(
+                        createPrefilledTestContext()
+                                .withOption(SINK_PARALLELISM.key(), "2")
+                                .build());
+        assertTrue(sink instanceof ElasticsearchDynamicSink);
+        ElasticsearchDynamicSink esSink = (ElasticsearchDynamicSink) sink;
+        SinkProvider provider =
+                (SinkProvider) esSink.getSinkRuntimeProvider(new SinkRuntimeProviderContext(false));
```

Comment:
Instead of using `SinkRuntimeProviderContext` we could move `MockContext` from `ElasticsearchDynamicSinkBase` to `ElasticsearchUtils` and use that one instead. In this case we don't need to add `flink-table-planner` as a dependency.